pkg/util/util.go

```diff
@@ -239,3 +243,123 @@ func SplitQualifiedName(str string) (string, string) {
 func JoinQualifiedName(namespace, name string) string {
 	return path.Join(namespace, name)
 }
+
+type Route struct {
+	Interface   string
+	Destination net.IP
+	Gateway     net.IP
+	// TODO: add more fields here if needed
+}
+
+func getRoutes(file io.Reader) ([]Route, error) {
+	routes := []Route{}
+	scanner := bufio.NewReader(file)
+	for {
+		line, err := scanner.ReadString('\n')
+		if strings.HasPrefix(line, "Destination") || strings.HasPrefix(line, "Kernel") {
+			continue
+		}
+		if err == io.EOF {
+			break
+		}
+		fields := strings.Fields(line)
+		if len(fields) < 8 {
+			return nil, fmt.Errorf("Incorrect number of fields (expected at least 8, got %d): %s", len(fields), line)
+		}
+		routes = append(routes, Route{})
+		route := &routes[len(routes)-1]
+		route.Interface = fields[7]
+		ip := net.ParseIP(fields[0])
+		if ip == nil {
+			return nil, fmt.Errorf("Unable to Parse IP: %s", fields[0])
+		}
+		route.Destination = ip
+		ip = net.ParseIP(fields[1])
+		if ip == nil {
+			return nil, fmt.Errorf("Unable to Parse IP: %s", fields[1])
+		}
+		route.Gateway = ip
+	}
+	return routes, nil
+}
+func getRouteTables() (string, error) {
+	args := []string{"-n"}
+	out, err := exec.Command("route", args...).Output()
+	if err != nil {
+		return "", err
+	}
+	return string(out), err
+}
+func ChooseHostInterface() (net.IP, bool, error) {
+	found := false
+	routedata, err := getRouteTables()
+	if err != nil {
+		fmt.Printf("ERROR:getRouteTables, %v", err)
+		return nil, false, err
+	}
+	r := strings.NewReader(routedata)
+	finalIP := net.ParseIP("127.0.0.1")
+	if finalIP == nil {
+		return nil, false, nil
+	}
+	routes, err := getRoutes(r)
+	if err != nil {
+		glog.Infof("ERROR:getRoutes, %v", err)
```

Comment:
debug?  Remove, or make more useful. (and remove "ERROR" since it's going to Info)